pyaedt/modeler/pcb/object3dlayout.py

```diff
@@ -1612,6 +1612,114 @@ def is_3d_placement(self):
         else:
             return False
 
+    @is_3d_placement.setter
+    def is_3d_placement(self, value):
+        props = ["NAME:3D Placement", "Value:=", value]
+        self.change_property(props)
+
+    @property
+    def is_flipped(self):
+        """Retrieve if the component is flipped or not."""
+        if self._oeditor.GetPropertyValue("BaseElementTab", self.name, "Flipped") in ["true", "True"]:
+            return True
+        else:
+            return False
+
+    @is_flipped.setter
+    def is_flipped(self, value):
+        props = ["NAME:Flipped", "Value:=", value]
+        self.change_property(props)
+
+    @property
+    def rotation_axis(self):
+        """Retrieve if the component is flipped or not."""
+        if self.is_3d_placement:
+            return self._oeditor.GetPropertyValue("BaseElementTab", self.name, "Rotation Axis")
+        return False
```

Comment:
I would raise an exception instead.
```suggestion
        raise ValueError("The component has no 3D placement.")
```

Reply:
the class should work both with 3dplacement flag enabled and disabled. it is an option, I'm not confident in raising Error